crates/sol-types/src/coder/encoder.rs

```diff
@@ -153,6 +128,31 @@ impl Encoder {
     pub fn append_head_tail<'a, T: TokenSeq<'a>>(&mut self, token: &T) {
         token.encode_sequence(self);
     }
+
+    /// Append a sequence of bytes, padding to the next word.
+    #[inline(always)]
+    fn append_bytes(&mut self, bytes: &[u8]) {
+        let n_words = utils::words_for(bytes);
+        self.buf.reserve(n_words);
+        unsafe {
+            // set length before copying
+            // this is fine because we reserved above and we don't panic below
+            let len = self.buf.len();
+            self.buf.set_len(len + n_words);
+
+            // copy
+            let cnt = bytes.len();
+            let dst = self.buf.as_mut_ptr().add(len).cast::<u8>();
+            ptr::copy_nonoverlapping(bytes.as_ptr(), dst, cnt);
+
+            // set remaining bytes to zero if necessary
+            let rem = cnt % 32;
+            if rem != 0 {
+                let pad = 32 - rem;
+                ptr::write_bytes(dst.add(cnt), 0, pad);
+            }
+        }
+    }
```

Comment:
`append_bytes` copied 32 bytes at a time, when it can just copy everything at once